posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -262,9 +263,18 @@ async def handle_model_ready(model: ModelNode, team_id: int, queue: asyncio.Queu
     try:
         if model.selected is True:
             team = await database_sync_to_async(Team.objects.get)(id=team_id)
-            await materialize_model(model.label, team)
+            # Get the workflow ID from the activity info
+            workflow_id = temporalio.activity.info().workflow_id
+            key, delta_table, job_id = await materialize_model(model.label, team, workflow_id)
     except Exception as err:
         await logger.aexception("Failed to materialize model %s due to error: %s", model.label, str(err))
+
+        if job_id:
```

Comment:
Yeah you'll need to check whether `job_id` is in `locals()` to make sure it's bound. e.g.
```python
if "job_id" in locals():
```